src/shell-app-system.c

```diff
@@ -113,13 +128,49 @@ add_aliases (ShellAppSystem  *self,
 {
   ShellAppSystemPrivate *priv = self->priv;
   const char *alias;
+  g_autofree char *renamed_from = NULL;
 
   alias = g_desktop_app_info_get_string (info, X_ENDLESS_ALIAS_KEY);
   if (alias != NULL)
     {
       char *desktop_alias = g_strconcat (alias, ".desktop", NULL);
       g_hash_table_insert (priv->alias_to_id, desktop_alias, g_strdup (id));
     }
+
+  /* We would like to call g_key_file_get_string_list(), but GDesktopAppInfo
+   * does not expose its internal GKeyFile, nor provides a get_string_list()
+   * wrapper. GKeyFile does not expose its string-list parser, so we open-code
+   * our own.
+   *
+   * An interesting quirk of the GKeyFile format is that lists are (supposed to
+   * be) ;-terminated, not just ;-delimited. This is so you can express an
+   * empty trailing element: "a;;" represents the list ["a", ""].
+   *
+   * It is possible to escape a semicolon in a list: "a\;b;" represents ["a;",
+   * "b"]. We cannot support this here, because g_key_file_get_string(), used
+   * internally by g_desktop_app_info_get_string(), rejects the "\;" sequence
+   * and returns NULL.
+   *
+   * https://gitlab.gnome.org/GNOME/glib/merge_requests/339
+   */
```

Comment:
This is a brilliant commit message, but not so much as a comment.

Reply:
I can move the bulk of it to the commit message, but I do really want to make a note of it in the code, in the hope that when we have GLib 2.60 with the function I wrote in that MR, we can update this code